.github/workflows/python-validate.yml

```diff
@@ -0,0 +1,76 @@
+name: "Python: Validate"
+
+on:
+  pull_request:
+    branches: [main]
+    paths:
+      - python/**
+      - proto/**
+
+jobs:
+  linuxPython37:
+    name: "[Linux] Python 3.7: Lint, Formatting & Unit Tests"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 30
+
+      - name: Retrieve last main commit (for `git diff` purposes)
+        run: |
+          git checkout -b pr
+          git fetch --prune --depth=30 origin +refs/heads/main:refs/remotes/origin/main
+          git checkout main
+          git checkout pr
+
+      - name: Resolve path filters
+        uses: dorny/paths-filter@v2
+        id: pathChanges
+        with:
+          filters: |
+            sdk:
+              - 'python/packages/sdk/**'
+            sdkSchema:
+              - 'python/packages/sdk-schema/**'
+            proto:
+              - 'proto/**'
+
+      - name: Install Python 3.7
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.7'
+
+          # ensure project dependencies are cached
+          # When using only `pyproject.toml` for dependencies, see:
+          #  https://github.com/actions/setup-python/issues/529#issuecomment-1367029699
+          cache: 'pip'
+          cache-dependency-path: |
+            **/pyproject.toml
+
+      - name: Install SDK project and dependencies
+        run: |
```

Comment:
Let's cover SDK (with deps install and unit tests) only if there are changes to `python/packages/sdk` (e.g. in Node.js we guard that with: https://github.com/serverless/console/blob/main/.github/workflows/node-validate.yml#L286

It'll help us to have CI more to the point.